SocialCareCaseViewerApi.Tests/V1/Controllers/CaseStatus/CaseStatusCreateTests.cs

```diff
@@ -0,0 +1,97 @@
+        // private readonly Fixture _fixture = new Fixture();
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockCaseStatusesUseCase = new Mock<ICaseStatusesUseCase>();
+
+            _caseStatusController = new CaseStatusController(_mockCaseStatusesUseCase.Object);
+        }
+
+
+        [Test]
+        public void CallsCaseStatusUseCaseToCreateCaseStatus()
+        {
+            _mockCaseStatusesUseCase.Setup(x => x.ExecutePost(It.IsAny<CreateCaseStatusRequest>()));
+            var request = CaseStatusHelper.CreateCaseStatusRequest();
+
+            var response = _caseStatusController.CreateCaseStatus(request) as BadRequestObjectResult;
+
+            _mockCaseStatusesUseCase.Verify(x => x.ExecutePost(request));
+        }
+
+        [Test]
+        public void WhenPersonNotFoundExceptionIsThrownReturns400WithMessage()
+        {
+            var exceptionMessage = "error message";
+            _mockCaseStatusesUseCase.Setup(x => x.ExecutePost(It.IsAny<CreateCaseStatusRequest>()))
+                .Throws(new PersonNotFoundException(exceptionMessage));
+
+            var request = CaseStatusHelper.CreateCaseStatusRequest();
+
+            var response = _caseStatusController.CreateCaseStatus(request) as BadRequestObjectResult;
+
+            response?.StatusCode.Should().Be(400);
+            response?.Value.Should().Be(exceptionMessage);
+        }
+
+        [Test]
+        public void WhenCaseTypeNotFoundExceptionIsThrownReturns400WithMessage()
+        {
+            var exceptionMessage = "error message";
+            _mockCaseStatusesUseCase.Setup(x => x.ExecutePost(It.IsAny<CreateCaseStatusRequest>()))
+                .Throws(new CaseStatusTypeNotFoundException(exceptionMessage));
+
+            var request = CaseStatusHelper.CreateCaseStatusRequest();
+
+            var response = _caseStatusController.CreateCaseStatus(request) as BadRequestObjectResult;
+
+            response?.StatusCode.Should().Be(400);
+            response?.Value.Should().Be(exceptionMessage);
+        }
+
+        [Test]
+        public void WhenCaseTypeAlreadyExistsExceptionIsThrownReturns400WithMessage()
+        {
+            var exceptionMessage = "error message";
+            _mockCaseStatusesUseCase.Setup(x => x.ExecutePost(It.IsAny<CreateCaseStatusRequest>()))
+                .Throws(new CaseStatusAlreadyExistsException(exceptionMessage));
+            var request = CaseStatusHelper.CreateCaseStatusRequest();
+
+            var response = _caseStatusController.CreateCaseStatus(request) as BadRequestObjectResult;
+
+            response?.StatusCode.Should().Be(400);
+            response?.Value.Should().Be(exceptionMessage);
+        }
+
+        [Test]
+        public void WhenRequestIsValidReturnsSuccessfulResponse()
+        {
+            var request = CaseStatusHelper.CreateCaseStatusRequest();
+            var response = _caseStatusController.CreateCaseStatus(request);
```

Comment:
Should we not be returning the response from mocking the useCase?

Then validate that when we call _caseStatusController.CreateCaseStatus(request) we return the 201 staus code and the value is the appropriate string?

Reply:
Functions now mocks the ExecutePost function to return a mocked case status + evaluates 201 and returned string